File: 프로그래머스/2/1844. 게임 맵 최단거리/게임 맵 최단거리.js
function inRange(x, y, maps){
    return 0 <= x && x < maps.length && 0 <= y && y < maps[0].length
}

function solution(maps) {
    var answer = 0;
    let dx = [1, 0, -1, 0]
    let dy = [0, 1, 0, -1]
    let visited = Array.from({length: maps.length}, () => Array(maps[0].length).fill(false))
    let q = []
    visited[0][0] = true
    q.push([0,0])
    while(q.length > 0){
        let [x, y] = q.shift()
        for(let i = 0; i < 4; i++){
            let nx = x + dx[i]
            let ny = y + dy[i]
            if(inRange(nx,ny,maps) && !visited[nx][ny] && maps[nx][ny] == 1){
                visited[nx][ny] = true
                maps[nx][ny] = maps[x][y] + 1
                q.push([nx,ny])
            }
        }
    }
    return maps[maps.length-1][maps[0].length - 1] > 1 ? maps[maps.length - 1][maps[0].length - 1] : -1
}

File: 프로그래머스/2/1844. 게임 맵 최단거리/게임 맵 최단거리.py
from collections import deque

def in_range(x, y, maps):
    return 0 <= x < len(maps) and 0 <= y < len(maps[0])

def solution(maps):
    answer = 0
    dx = [0, 1, 0, -1]
    dy = [1, 0, -1, 0]
    visited = [[False] * len(maps[0]) for _ in range(len(maps))]
    q = deque()
    start = [0, 0]
    q.append(start)
    visited[start[0]][start[1]] = True
    while q:
        x, y = q.popleft()
        for i in range(4):
            nx, ny = x + dx[i], y + dy[i]
            if in_range(nx, ny, maps) and not visited[nx][ny] and maps[nx][ny] == 1:
                visited[nx][ny] = True
                maps[nx][ny] = maps[x][y] + 1
                q.append([nx, ny])
    return maps[-1][-1] if maps[-1][-1] > 1 else -1